datastacks/datastacks_cli.py

```diff
@@ -0,0 +1,38 @@
+import logging
+import shutil
+import yaml
+
+import click
+from click_loglevel import LogLevel
+
+from datastacks.logger import setup_logger
+from datastacks.utils import render_template_component
+
+
+@click.group()
+@click.option("--log-level", "-l", type=LogLevel(), default=logging.INFO)
+def cli(log_level):
+    setup_logger("datastacks", log_level)
+
+
+@click.command()
+@click.option("--config-path", "-c", type=str, help="Absolute path to config file on local machine")
+@click.option("--data-quality/--no-data-quality", "-dq/", default=False, help="Flag to determine whether to include data quality in template")
+def gen_ingest(config_path, data_quality):
+    """Generate new ingest pipeline"""
+    if data_quality:
+        template_source_folder = "Ingest_SourceType_SourceName_DQ"
```

Comment:
Just another thought... this logic might need to be slightly different, depending on the final structure of the templates.
I think (@motylp to confirm) for the DQ option we'll need to:

1. Load in the templates from the regular (non-DQ) path
2. Load in from the DQ path, overwriting any files with the same name

Reply:
I thought we'd agreed to not pull out the common files for now? As much as it might make the templates folder tidier it does make things more complex here

Reply:
I think there are two topics here:
- pulling out the common files that we don't expect to vary between all pipelines (ingest, silver, gold etc.), e.g. `provider.tf`, `constraints.tf`, (maybe) `data.tf`. That we agreed not to do.
- For DQ and non-DQ in Ingest within templates PR @adurkan-amido  suggested to separate DQ-related files (although we initially agreed otherwise), so that we don't duplicate. And I agreed to do it, since working with two versions of the same files was really annoying for me. But since it's all done now, I can just copy over what's missing from DQ. Both options have pros and cons, but I guess since we agreed to use jinja suffixes for maintenance simplification, adding this logic to overwrite what's missing now, might be an overkill :) I can duplicate this back. I don't have a strong preference here.

Reply:
I lean towards the duplicated version as otherwise we likely want to rethink how we're handling the templating overall, what do you reckon @adurkan-amido

Reply:
I think if we did one pass of `render_template_components` for the `Ingest_SourceType_SourceName`, then another pass of it for `Ingest_SourceType_SourceName_DQ` if DQ is specified, that would handle it? To me that seems quite simple and means we don't have 2 versions of the same files to maintain. If I'm missing something and it's going to be more complicated than that, happy to go for the other approach.

Reply:
It's just that we're generally avoiding deleting files for now. I'm happy to match what we've done for templating though. Did we leave that as duplicated?

Reply:
We left is as non-duplicated for now and if decided otherwise, the files can be re-added within this or a separate PR.

Reply:
Cool I'll make the changes.

Reply:
Done